core/vm/contracts_write_test.go

```diff
@@ -211,3 +199,92 @@ var StakingPrecompileTests = []writeCapablePrecompileTest{
 	//	name:          "migrationAddressMismatch",
 	//},
 }
+
+func TestStakingPrecompiles(t *testing.T) {
+	for _, test := range StakingPrecompileTests {
+		testStakingPrecompile(test, t)
+	}
+}
+
+func TestWriteCapablePrecompilesReadOnly(t *testing.T) {
+	p := &stakingPrecompile{}
+	expectedError := errWriteProtection
+	res, err := RunWriteCapablePrecompiledContract(p, nil, nil, []byte{}, true)
+	if err != nil {
+		if err.Error() != expectedError.Error() {
+			t.Errorf("Expected error %v, got %v", expectedError, err)
+		}
+	} else {
+		t.Errorf("Expected an error %v but instead got result %v", expectedError, res)
+	}
+}
+
+func transfer(db StateDB, sender, recipient common.Address, amount *big.Int, txType types.TransactionType) {
+}
+
+func testCrossShardXferPrecompile(test writeCapablePrecompileTest, t *testing.T) {
+	// this EVM needs stateDB, Transfer, and NumShards
+	var db ethdb.Database
+	var err error
+	defer func() {
+		if db != nil {
+			db.Close()
+		}
+	}()
+	if db, err = rawdb.NewLevelDBDatabase("/tmp/harmony_shard_0", 256, 1024, ""); err != nil {
+		db = nil
+		t.Fatalf("Could not initialize db %s", err)
+	}
+	stateCache := state.NewDatabase(db)
+	state, err := state.New(common.Hash{}, stateCache)
+	if err != nil {
+		t.Fatalf("Error while initializing state %s", err)
+	}
+	var env = NewEVM(Context{
+		NumShards: 4,
+		Transfer:  transfer,
+	}, state, params.TestChainConfig, Config{})
+	p := &crossShardXferPrecompile{}
+	testWriteCapablePrecompile(test, t, env, p)
+}
+
+func TestCrossShardXferPrecompile(t *testing.T) {
+	for _, test := range CrossShardXferPrecompileTests {
+		testCrossShardXferPrecompile(test, t)
+	}
+}
+
+var CrossShardXferPrecompileTests = []writeCapablePrecompileTest{
+	{
+		input:    []byte{40, 72, 8, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5, 107, 199, 94, 45, 99, 16, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 165, 36, 21, 19, 218, 159, 68, 99, 241, 212, 135, 75, 84, 141, 251, 172, 41, 217, 31, 52, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1},
```

Comment:
does the test cases cover invalid values that could cause errors or failures? We need to make sure the new code won't cause evm or system fault.